src/cmap/connect.ts

```diff
@@ -331,12 +350,27 @@ function makeConnection(options: ConnectionOptions, _callback: CallbackWithType<
     _callback(err, ret);
   };
 
+  if (proxyOptions?.host != null) {
+    // Currently, only Socks5 is supported.
+    return makeSocks5Connection(
+      {
+        ...options,
+        connectTimeoutMS: connectionTimeout, // Should always be present for Socks5
+        proxyOptions: undefined
+      },
+      proxyOptions,
+      callback
+    );
+  }
+
   if (useTLS) {
     const tlsSocket = tls.connect(parseSslOptions(options));
     if (typeof tlsSocket.disableRenegotiation === 'function') {
       tlsSocket.disableRenegotiation();
     }
     socket = tlsSocket;
+  } else if (existingSocket) {
```

Comment:
That wouldn’t quite work, because we also use `existingSocket` in the TLS case, it just isn’t quite obvious because that happens inside of `parseSslOptions` – added a comment about that :)